src/Sylius/Bundle/CoreBundle/Validator/Constraints/OrderPaymentMethodEligibilityValidator.php

```diff
@@ -14,12 +14,21 @@
 namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
 
 use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Payment\Checker\OrderPaymentMethodEligibilityCheckerInterface;
+use Sylius\Component\Core\Payment\Checker\OrderPaymentMethodPerChannelEligibilityCheckerInterface;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 use Webmozart\Assert\Assert;
 
 final class OrderPaymentMethodEligibilityValidator extends ConstraintValidator
 {
+    public function __construct(
+        private readonly OrderPaymentMethodEligibilityCheckerInterface $checker,
+        private readonly OrderPaymentMethodPerChannelEligibilityCheckerInterface $perChannelChecker,
+    ) {
+    }
```

Comment:
I was more thinking of creating separate Validator  to avoid addition of constructor to this one 🤔
So like creating for example `OrderPaymentMethodEnabledInChannelValidator`